barretenberg/cpp/src/barretenberg/vm2/simulation/events/bytecode_events.hpp

```diff
@@ -23,7 +23,8 @@ struct BytecodeDecompositionEvent {
 
 struct BytecodeHashingEvent {
     BytecodeId bytecode_id;
-    std::shared_ptr<std::vector<uint8_t>> bytecode;
+    uint32_t bytecode_length;
```

Comment:
I think you don't use the length?

Reply:
Or at least you are not setting it in simulation

Reply:
good catch - added a tracegen test as well